src/commcare_cloud/ansible/restart_webworker_services.yml

```diff
@@ -1,6 +1,6 @@
 - name: Rolling restart webworker services
   hosts: webworkers
```

Comment:
Amazing find, @mkangia and @AmitPhulera!

I think the best solution would be to break this up into two groups like you noted @mkangia, so we hav one with `hq_webworkers` and one with `mobile_webworkers`. To make matters more complicated, it would need to detect whether that split exists (as in SaaS production for example) or we just have one big `webworkers` group (as possibly in some self-hosted setups).

I think we could achieve that by passing hosts (e.g. `webworker_hosts`) as a variable into this playbook and move this logic one level up to https://github.com/dimagi/commcare-cloud/blob/f66001d7386ea30437e99a2093415f2a31d6702b/src/commcare_cloud/ansible/restart_commcare_services.yml#L28 with something like this:

```yaml
# here we use groups['hq_webworkers'] being truthy or not
# to determine whether webworkers are split (into hq_webworkers versus mobile_webworkers) or not

- name: "Restart all webworkers (if not split)"
  import_playbook: restart_webworker_services.yml
  vars:
    webworker_hosts: 'webworkers'
  when: not groups['hq_webworkers']

- name: "Restart hq_webworkers (if split)"
  import_playbook: restart_webworker_services.yml
  vars:
    webworker_hosts: 'hq_webworkers'
  when: groups['hq_webworkers']

- name: "Restart mobile_webworkers (if split)"
  import_playbook: restart_webworker_services.yml
  vars:
    webworker_hosts: 'mobile_webworkers'
  when: groups['hq_webworkers']
```

I haven't tested this, but in theory something like that should work well. (I'm often wrong about what ansible constructs are available in what contexts, so that's an example of something I could have gotten wrong!)

Reply:
Thanks @dannyroberts, good to see that something like this is possible.
I am okay with holding off merging this PR if someone should give that solution a try. I believe we are okay till next deploy is needed on prod.

Reply:
I think if this is a meaningful stop-gap measure we should go ahead and merge it, to give any deploy in the interim the best chance of succeeding. I'll go ahead and do that and then also make a note to consider reverting once we have the "better" solution.

Reply:
Cool. Thanks for the merge @dannyroberts